src/world/scheduler/builder.rs

```diff
@@ -189,12 +188,14 @@ impl WorkloadBuilder {
             }
         }
 
-        self.systems.push((
-            TypeId::of::<S>(),
-            type_name::<F>(),
-            old_len..self.borrow_info.len(),
-            Box::new(system),
-        ));
+        let workload_system = WorkloadSystem {
+            borrow_constraints: borrows,
+            system_fn: Box::new(system),
+            system_type_id: TypeId::of::<S>(),
+            system_type_name: type_name::<F>(),
+        };
+
+        self.systems.push(workload_system);
 
         Ok(self)
```

Comment:
My idea is that we could reduce `try_with_system` to only `self.systems.push(workload_system); Ok(self)`, then make `WorkloadSystem` a result of `system!(system_fn)`. I believe that this will be great for additional shipyard tooling like I'm working on with Storyscript, because I won't need to resort to additional generic complexity if I want to collect the systems in a Vec or something in a separate step from actually building the Workload.